tests/integration/tls/test_manual_tls.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+import pytest
+from juju.application import Application
+from pytest_operator.plugin import OpsTest
+
+from ..helpers import APP_NAME, MODEL_CONFIG, SERIES, UNIT_IDS
+from ..helpers_deployments import wait_until
+from .helpers_manual_tls import MANUAL_TLS_CERTIFICATES_APP_NAME, ManualTLSAgent
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.runner(["self-hosted", "linux", "X64", "jammy", "large"])
+@pytest.mark.group(1)
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+async def test_build_and_deploy_with_manual_tls(ops_test: OpsTest) -> None:
+    """Build and deploy prod cluster of OpenSearch with Manual TLS Operator integration."""
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.set_config(MODEL_CONFIG)
+
+    os_app: Application = await ops_test.model.deploy(
+        my_charm,
+        num_units=len(UNIT_IDS),
+        series=SERIES,
+        application_name=APP_NAME,
+    )
+
+    # Deploy TLS Certificates operator.
+    tls_app: Application = await ops_test.model.deploy(
+        MANUAL_TLS_CERTIFICATES_APP_NAME,
+        channel="beta",
```

Comment:
What's the reason why you chose to test against the `beta` channel? Most other tests use `stable` for the channel of the `self-signed-certificates`, I think we should stick to this if no specific reason for `beta`.

Reply:
TBH I just followed the [readme](https://github.com/canonical/manual-tls-certificates-operator) they had. There is no specific reason. I will test with the stable version, there shouldn't be any issue.

Reply:
They have both `stable` and `beta` as the same revision. I changed the channel to `stable`. I'm closing the conversation.